src/System.Runtime.Extensions/src/System/Net/WebUtility.cs

```diff
@@ -425,7 +427,7 @@ private static string UrlDecodeInternal(string value, Encoding encoding)
                     helper.AddChar(ch);
```

Comment:
I don't understand how this AddByte vs AddChar mechanism works.  It looks like GetString dumps all of the bytes to the end of the chars... doesn't that result in bytes that came in the middle of the input getting decoded out of order?